File: src/co/edu/uptc/music/servlets/GenresServlet.java
package co.edu.uptc.music.servlets;

import java.io.IOException;

import com.google.gson.Gson;

import java.io.PrintWriter;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import co.edu.uptc.music.logic.managers.GenresManager;

@WebServlet(name = "GenresServlet", urlPatterns = {"/GenresServlet"})
public class GenresServlet extends HttpServlet {

    GenresManager genresManager = new GenresManager();

    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
            throws Exception {
        response.setContentType("application/json");
        response.setCharacterEncoding("UTF-8");
        Gson gson = new Gson();
        String data = request.getParameter("data");
        try (PrintWriter writer = response.getWriter()) {
            genresManager.loadGenres();
            if (genresManager.getList().size() > 0) {
                writer.print("{\"genres\":" + gson.toJson(genresManager.getList()) + "}");
            }
            writer.close();
        }
    }

    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws
            ServletException, IOException {
        try {
            processRequest(req, resp);
        } catch (Exception ignored) {
        }
    }

    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws
            ServletException, IOException {

        try {
            processRequest(req, resp);
        } catch (Exception ignored) {
        }
    }

    @Override
    public String getServletInfo() {
        return "Genre servlet";
    }


}

File: src/co/soundwave/servlets/GenresServlet.kt
package co.soundwave.servlets

import co.soundwave.extensions.ignore
import co.soundwave.managers.GenresManager
import com.google.gson.Gson
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@WebServlet(name = "GenresServlet", urlPatterns = ["/GenresServlet"])
class GenresServlet : BaseServlet() {
    
    private val genresManager = GenresManager()
    
    override fun processRequest(
        request: HttpServletRequest,
        response: HttpServletResponse,
        type: Type
                               ) {
        ignore {
            response.contentType = "application/json"
            response.characterEncoding = "UTF-8"
            val gson = Gson()
            val data = request.getParameter("data")
            response.writer.use { writer ->
                genresManager.load()
                if (genresManager.getList().size > 0) {
                    writer.print("{\"genres\":" + gson.toJson(genresManager.getList()) + "}")
                }
                writer.close()
            }
        }
    }
    
    override fun getServletInfo(): String = "Genre servlet"
}
